apps/teams-test-app/src/public/naa_childIframe.html

```diff
@@ -0,0 +1,219 @@
+      </div>
+    </div>
+
+    <script>
+      const targetOrigin = 'https://local.teams.office.com:8080';
+
+      // Utility function to display a response in a given element
+      const displayResponse = (element, message) => {
+        element.innerText = message;
+      };
+
+      // Predefined payloads
+      const defaultPayloadForBridge =
+        '{"messageType":"NestedAppAuthRequest","method":"GetToken","sendTime":1732269811006,"clientLibrary":"testOS","clientLibraryVersion":"1.0.0","requestId":"684352c2-7ab7-4def-b7e1-XXXXXXXXXXX","tokenParams":{"correlationId":"39dc85fe-9054-11ed-a1eb-XXXXXXXXXXX"}}';
+      const defaultPayloadForTopWindow =
+        '{"id":"2","func":"nestedAppAuth.execute","args":[],"data":"{\\"messageType\\":\\"NestedAppAuthRequest\\",\\"method\\":\\"GetToken\\",\\"sendTime\\":1732269811006,\\"clientLibrary\\":\\"testOS\\",\\"clientLibraryVersion\\":\\"1.0.0\\",\\"requestId\\":\\"684352c2-7ab7-4def-b7e1-XXXXXXXXXXX\\",\\"tokenParams\\":{\\"correlationId\\":\\"39dc85fe-9054-11ed-a1eb-XXXXXXXXXXX\\"}}"}';
+
+      // Functions to set default payloads
+      function setDefaultPayloadForBridge() {
+        const inputElement = document.getElementById('input_childPayloadForBridge');
+        if (inputElement) {
+          inputElement.value = defaultPayloadForBridge;
+        }
+      }
+
+      function setDefaultPayloadForTopWindow() {
+        const inputElement = document.getElementById('input_childPayloadForTopWindow');
+        if (inputElement) {
+          inputElement.value = defaultPayloadForTopWindow;
+        }
+      }
+
+      // Send Message to NestedAppAuthBridge
+      function sendMessageToBridge() {
+        const inputElement = document.getElementById('input_childPayloadForBridge');
+        const responseElement = document.getElementById('text_bridge_response_for_child');
+
+        if (!inputElement || !responseElement) return;
+
+        try {
+          // Validate and send the payload
+          const payload = JSON.parse(inputElement.value);
+
+          // Inline validation for NestedAppAuthRequest
+          if (!payload) throw new Error('Input is required.');
+          if (payload.messageType !== 'NestedAppAuthRequest') {
+            throw new Error('Invalid or missing messageType. Expected "NestedAppAuthRequest".');
+          }
+          if (!payload.method) throw new Error('Method name is required in payload.');
+          if (!payload.requestId) throw new Error('RequestId is required in payload.');
+
+          const bridge = window.nestedAppAuthBridge;
+          if (!bridge) {
+            displayResponse(responseElement, 'NAA Bridge not available');
+            return;
+          }
+
+          bridge.postMessage(JSON.stringify(payload));
+          displayResponse(responseElement, 'Message sent to Bridge. Awaiting response...');
+
+          bridge.addEventListener('message', (response) => {
```

Comment:
Is there any risk of missing the response if we wait to add the event listener after posting the message to the NAA bridge?

Reply:
That's a good callout. I will move the addEventListener before the postMessage call. Thanks